src/integrationTest/java/com/mongodb/hibernate/IdentifierIntegrationTests.java

```diff
@@ -0,0 +1,250 @@
+                        .append(EndingWithSingleQuote.FIELD_NAME, new BsonInt32(item.v)));
+        sessionFactoryScope.inTransaction(session -> session.get(EndingWithSingleQuote.class, item.id));
+    }
+
+    @Test
+    void inDoubleQuotesMixedCase() {
+        var item = new InDoubleQuotesMixedCase();
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollectionInDoubleQuotesMixedCase.find())
+                .containsExactly(new BsonDocument()
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id))
+                        .append(InDoubleQuotesMixedCase.ACTUAL_FIELD_NAME, new BsonInt32(item.v)));
+        sessionFactoryScope.inTransaction(session -> session.get(InDoubleQuotesMixedCase.class, item.id));
+    }
+
+    @Test
+    void startingWithDoubleQuote() {
+        var item = new StartingWithDoubleQuote();
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollectionStartingWithDoubleQuote.find())
+                .containsExactly(new BsonDocument()
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id))
+                        .append(StartingWithDoubleQuote.FIELD_NAME, new BsonInt32(item.v)));
+        sessionFactoryScope.inTransaction(session -> session.get(StartingWithDoubleQuote.class, item.id));
+    }
+
+    @Test
+    void endingWithDoubleQuote() {
+        var item = new EndingWithDoubleQuote();
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollectionEndingWithDoubleQuote.find())
+                .containsExactly(new BsonDocument()
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id))
+                        .append(EndingWithDoubleQuote.FIELD_NAME, new BsonInt32(item.v)));
+        sessionFactoryScope.inTransaction(session -> session.get(EndingWithDoubleQuote.class, item.id));
+    }
+
+    @Override
+    public void injectSessionFactoryScope(SessionFactoryScope sessionFactoryScope) {
+        this.sessionFactoryScope = sessionFactoryScope;
+    }
+
+    @Entity
+    @Table(name = WithSpaceAndDotMixedCase.COLLECTION_NAME)
+    static class WithSpaceAndDotMixedCase {
+        static final String COLLECTION_NAME = "collection with space and .dot Mixed Case";
+        static final String FIELD_NAME = "field with space Mixed Case";
+
+        @Id
+        int id;
+
+        @Column(name = WithSpaceAndDotMixedCase.FIELD_NAME)
+        int v;
+    }
+
+    @Entity
+    @Table(name = InSingleQuotesMixedCase.COLLECTION_NAME)
+    static class InSingleQuotesMixedCase {
+        static final String COLLECTION_NAME = "`collection in single quotes Mixed Case`";
+        static final String FIELD_NAME = "`field in single quotes Mixed Case`";
+        static final String ACTUAL_COLLECTION_NAME = "collection in single quotes Mixed Case";
+        static final String ACTUAL_FIELD_NAME = "field in single quotes Mixed Case";
```

Comment:
yeah, Hibernate has its own quotes so the above ACTUAL IDENTIFIER makes sense, as long as the quote pairs surround the whole identifier. For completeness, we could cover `[]` as well.